pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadDownloadRestletResource.java

```diff
@@ -386,11 +386,11 @@ private SegmentMetadata getSegmentMetadata(File tempDecryptedFile, File tempSegm
   }
 
   private void completeZkOperations(boolean enableParallelPushProtection, HttpHeaders headers, File uploadedSegmentFile,
-      String tableNameWithType, SegmentMetadata segmentMetadata, String segmentName, String zkDownloadURI,
-      boolean moveSegmentToFinalLocation, String crypter)
+      String rawTableName, String tableNameWithType, SegmentMetadata segmentMetadata, String segmentName,
+      String zkDownloadURI, boolean moveSegmentToFinalLocation, String crypter)
       throws Exception {
     URI finalSegmentLocationURI = URIUtils
-        .getUri(ControllerFilePathProvider.getInstance().getDataDirURI().toString(), tableNameWithType,
+        .getUri(ControllerFilePathProvider.getInstance().getDataDirURI().toString(), rawTableName,
             URIUtils.encode(segmentName));
     ZKOperator zkOperator = new ZKOperator(_pinotHelixResourceManager, _controllerConf, _controllerMetrics);
     zkOperator.completeSegmentOperations(tableNameWithType, segmentMetadata, finalSegmentLocationURI, uploadedSegmentFile,
```

Comment:
If you check the previous PR for `completeSegmentOperations()`, there are many parts where we change `rawTableName -> tableNameWithType`. Those changes may need to be reverted as well?

e.g.
https://github.com/apache/incubator-pinot/pull/6567/files#diff-ab01aaed27b0a00398747217dea97c6ae4e13a1e51682c93da43375e87d15761L73

Or, is the fix in the PR sufficient? @chenboat probably has more context around this.

Reply:
Before my PR, most places already use tablename with type (in fact they use offline table name with type). As an example, 

`String offlineTableName = TableNameBuilder.OFFLINE.tableNameWithType(rawTableName);`

My changes are making the names more clear and generic. Of course, the deep store uri change one is not compatible -- I missed the segment deletion path.

Reply:
I checked all of those and I believe the changes in this PR are sufficient. Will wait for Ting's comments.

Reply:
thanks @sajjad-moradi . I did another walk through and arrived at the same conclusion.